deps/v8/src/arm/assembler-arm.cc

```diff
@@ -749,7 +778,7 @@ static bool fits_shifter(uint32_t imm32,
                          Instr* instr) {
   // imm32 must be unsigned.
   for (int rot = 0; rot < 16; rot++) {
-    uint32_t imm8 = (imm32 << 2*rot) | (imm32 >> (32 - 2*rot));
+    uint32_t imm8 = rot == 0 ? imm32 : ((imm32 << 2*rot) | (imm32 >> (32 - 2*rot)));
```

Comment:
This is an upstream patch (we are using clang on FreeBSD):
https://codereview.chromium.org/979633002/